### PR TITLE
feat!: bump rust-polars to 0.30.0

### PR DESCRIPTION
**Pull request by eitsupi**
https://github.com/pola-rs/polars/releases/tag/rs-0.29.0
https://github.com/pola-rs/polars/releases/tag/rs-0.30.0

### CODE REVIEW

**Comment by eitsupi** [comment]
The errors in Rust have been fixed but have not yet passed the R tests.
Feel free to edit this branch as I may not have enough time to work on this.

**Comment by etiennebacher** [comment]
I think this should close #230

**Comment by eitsupi** [comment]
I fixed as many of the test failures as I could.

The remaining failures are a problem where `date_range_lazy` now returns a list, so when converted to R, it becomes a list instead of a vector. (pola-rs/polars#8513)

**Comment by sorhawell** [comment]
Thanks. I will take a look at it.

**Comment by eitsupi** [comment]
I was just looking at this myself and Python's `date_range` behaves quite strangely.
I mean, when `eager=True` it returns an un-nested Series, but when `eager=False` it returns a list type, which I think is more appropriate here given the meaning of range, but anyway it returns something completely different for `True` and `Flase`.
Some tests must be removed if this is to be mimicked.

**Comment by eitsupi** [comment]
> I found polars did a behavior change on date_range such that it produces Eager = False a list of DateRanges.
> It is being discussed here (pola-rs/polars#9019)

Oh, thank you for finding that.

The change definitely makes sense, but we cannot incorporate that change here because it has not yet been made in the main branch.
Perhaps we can change the `lazy=FALSE` behavior in this repository using `implode`? (This is different from the behavior of py-polars, but since it will certainly be changed in py-polars in the future, it does not seem to be a problem to change it ahead of time.)

**Comment by sorhawell** [comment]
One timezone error remaining and the we have restore previous behavior.

We can update the date_range(s) behavior in a new PR after this has merged

**Comment by sorhawell** [comment]
tiny detail for reference "[fix subtle typo bug](https://github.com/pola-rs/r-polars/pull/289/commits/28c1b7e46c87420e562b840c3d94d5b2ff469e97)" is an incorrect commit msg. There was a unrelated subtle bug (high/low) which did not show, and then I disabled a test on timezone until a future refactor.

**Comment by eitsupi** [comment]
All tests now pass; I would like to defer the renaming from `list` to `arr` on the R side to a follow-up fix as it would be an additional major task.

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 - Replace the argument `reverse` by `descending` in all sorting functions. This
   is for consistency with the upstream Polars (#291, #293).
+- Bump rust-polars from 2023-04-20 unreleased version to version 0.30.0 released in 2023-05-30 (#289).
+    - Rename `concat_lst` to `concat_list`.
+    <!-- TODO: - Rename `arr` to `list`. -->
+    - Rename `$str$explode` to `$str$str_explode`.
+    - Remove `tz_aware` and `utc` arguments from `str_parse`.
+    - `$date_range`'s the `lazy` argument is default to `TRUE`.
```
Comment (etiennebacher) [inline NEWS.md:12]:
```suggestion
    - in `$date_range`'s the `lazy` argument is now `TRUE` by default.
```

File: R/expr__string.R [hunk 2]
```diff
@@ -706,20 +700,20 @@ ExprStr_slice = function(offset, length = NULL) {
     unwrap("in str$slice:")
 }
 
-#' explode
-#' @name ExprStr_explode
+#' str_explode
```
Comment (etiennebacher) [inline R/expr__string.R:703]: Would be better to have a proper title here (not just for this function)

**Review by etiennebacher — APPROVED** [review]
Looks great, thanks for updating this. It's fine for me to do all the renaming from `arr` to `list` in a separate PR, are you also ok with this @sorhawell ?

**Comment by eitsupi** [comment]
@etiennebacher Thanks for your review.

Since this PR contains updates to the toolchain and everything needs to be recompiled every time we switch branches, I think it's better to merge it in sooner rather than later, so I'm merging it.
Additional modifications can be made later.......

**Pull request merged**